app/services/product_tag_rules_filterer.rb

```diff
@@ -0,0 +1,107 @@
+class ProductTagRulesFilterer
```

Comment:
We need a new spec and some tests services/product_tag_rules_filterer_spec

Reply:
Already added :+1: